edge/pkg/edgehub/controller.go

```diff
@@ -225,7 +225,9 @@ func (ehc *Controller) routeToCloud() {
 		}
 
 		// post message to cloud hub
+		ehc.keeperLock.Lock()
 		err = ehc.sendToCloud(message)
```

Comment:
You may move the lock/unlock operation into sendToCloud, and replace `err := ehc.chClient.Send(*msg)` in `keepalive()` with `err := ehc. sendToCloud (*msg)`

Reply:
Done.